playbooks/sample-sap-vm-provision-redhat-ocpv.yml

```
@@ -18,99 +18,40 @@
-
-        - name: Set predefined CA file
-          ansible.builtin.set_fact:
-            __sap_vm_provision_kubevirt_vm_register_ca_cert: "{{ sap_vm_provision_kubevirt_vm_ca_cert }}"
-
-    - name: Log into Red Hat OpenShift cluster (obtain access token)
-      community.okd.openshift_auth:
-        host: "{{ __sap_vm_provision_kubevirt_vm_register_api_endpoint }}"
-        username: "{{ sap_vm_provision_kubevirt_vm_admin_username }}"
-        password: "{{ sap_vm_provision_kubevirt_vm_admin_password }}"
-        ca_cert: "{{ __sap_vm_provision_kubevirt_vm_register_ca_cert }}"
-      register: __sap_vm_provision_kubevirt_vm_register_kubevirt_vm_auth_results
-
-    - name: Set token in kubeconfig
+    - name: Use kubeconfig file specified in environment variable K8S_AUTH_KUBECONFIG | KUBECONFIG if sap_vm_provision_kubevirt_vm_kubeconfig_path is not defined
+      when: >
+        sap_vm_provision_kubevirt_vm_kubeconfig is not defined or
+        sap_vm_provision_kubevirt_vm_kubeconfig == None or
+        sap_vm_provision_kubevirt_vm_kubeconfig == ''
       ansible.builtin.set_fact:
-        __sap_vm_provision_kubevirt_vm_register_kubeconfig_data: >-
-          {{
-            __sap_vm_provision_kubevirt_vm_register_kubeconfig_data | combine({
-              'users':  __sap_vm_provision_kubevirt_vm_register_kubeconfig_data.users | map('combine', [{'user': {'token': __sap_vm_provision_kubevirt_vm_register_kubevirt_vm_auth_results.openshift_auth.api_key }}] )
-            }, recursive=True)
-          }}
-
-    - name: Write the updated kubeconfig
-      ansible.builtin.copy:
-        content: "{{ __sap_vm_provision_kubevirt_vm_register_kubeconfig_data | to_nice_yaml }}"
-        dest: "{{ __sap_vm_provision_kubevirt_vm_register_kubeconfig }}"
-        mode: "0600"
+         sap_vm_provision_kubevirt_vm_kubeconfig: "{{ lookup('env', 'K8S_AUTH_KUBECONFIG') | default(lookup('env', 'KUBECONFIG'), true) }}"
```

ditto. should we fail if both are missing?